test/test_datapipe.py

```diff
@@ -862,6 +864,11 @@ def test_fork_datapipe(self):
         self.assertEqual(len(input_dp), len(dp2))
         self.assertEqual(len(input_dp), len(dp3))
 
+        # Pickle Test:
+        dp1, dp2, dp3 = input_dp.fork(num_instances=3)
+        from torch.utils.data.graph import traverse
+        traverse(dp1)  # This should not raise any error
```

Comment:
Could we add another test that we iterates `dp1`, `dp2` and `dp3` and then traverse?

Reply:
Done